tests/p2p_connection_test.rs

```diff
@@ -6,46 +6,124 @@ mod setup;
 #[cfg(feature = "nigiri")]
 mod p2p_connection_test {
     use super::*;
+    use std::thread::sleep;
+    use std::time::Duration;
 
     use crate::setup::NodeHandle;
     use uniffi_lipalightninglib::config::NodeAddress;
 
+    const NIGIRI_LND_ADDR: &str = "127.0.0.1:9735";
+    const FAULTY_ADDR: &str = "127.0.0.1:9";
+
     #[test]
     fn test_successful_p2p_connection() {
         setup::nigiri::start();
         let lsp_info = setup::nigiri::query_lnd_info().unwrap();
         let lsp_node = NodeAddress {
             pub_key: lsp_info.pub_key,
-            address: "127.0.0.1:9735".to_string(),
+            address: NIGIRI_LND_ADDR.to_string(),
         };
 
-        let node = NodeHandle::new(lsp_node).start().unwrap();
-        assert_eq!(node.get_node_info().num_peers, 1)
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
+        assert_eq!(node.get_node_info().num_peers, 1);
+        assert!(node.connected_to_node(&lsp_node));
     }
 
     #[test]
     fn test_failing_p2p_connection() {
+        setup::nigiri::start();
         let lsp_info = setup::nigiri::query_lnd_info().unwrap();
         let lsp_node = NodeAddress {
             pub_key: lsp_info.pub_key,
-            address: "127.0.0.1:9".to_string(),
+            address: FAULTY_ADDR.to_string(),
         };
-        assert!(NodeHandle::new(lsp_node).start().is_err());
+
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
+        assert_eq!(node.get_node_info().num_peers, 0);
+        assert!(!node.connected_to_node(&lsp_node));
+    }
+
+    #[test]
+    fn test_start_node_while_lsp_is_down() {
+        let lsp_info = setup::nigiri::query_lnd_info().unwrap();
+        let lsp_node = NodeAddress {
+            pub_key: lsp_info.pub_key,
+            address: NIGIRI_LND_ADDR.to_string(),
+        };
+
+        // Let's shutdown LND, while we leave Esplora running (test should not affect chain sync)
+        setup::nigiri::pause();
+        setup::nigiri::resume_without_ln();
+
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+
+        // Wait for the LDK to connect to the LSP (which should fail)
+        sleep(Duration::from_millis(1500));
```

Comment:
We can remove the `sleep()` as it's already being done inside `start()` right?

Reply:
The other option is to remove the sleep from `start()`. Maybe this latter option is best because we'll need to sleep anyway after restarting nigiri and it may get confusing when a sleep is needed and when it's not.

Reply:
The way I understand it, is that the `start()` function waits until Nigiri is up and running. However, that does not necessarily mean that LDK already reconnected to the now started LND instance. Remember that it retries every second.

That given, I think both the `sleep` within `start()` and the `sleep` being discussed here have their own importance and should not be removed.